src/rules/selector-complexity.js

```diff
@@ -0,0 +1,526 @@
+
+		messages: {
+			maxSelectors:
+				"Exceeded maximum {{selector}} selector. Only {{limit}} allowed.",
+			disallowedSelectors:
+				"'{{selectorName}}' {{selector}} is not allowed.",
+		},
+	},
+
+	create(context) {
+		const [
+			{
+				maxIds,
+				maxClasses,
+				maxTypes,
+				maxAttributes,
+				maxPseudoClasses,
+				maxUniversals,
+				maxCompounds,
+				maxCombinators,
+				disallowCombinators,
+				disallowPseudoClasses,
+				disallowPseudoElements,
+				disallowAttributes,
+				disallowAttributeMatchers,
+			},
+		] = context.options;
+
+		return {
+			Selector(node) {
+				const selectors = node.children;
+				const selectorLoc = node.loc;
+
+				const idSelectors = getSelectors(selectors, "IdSelector");
+				const classSelectors = getSelectors(selectors, "ClassSelector");
+				const typeSelectors = selectors.filter(
+					child =>
+						child.type === "TypeSelector" && child.name !== "*",
+				);
+				const attributeSelectors = getSelectors(
+					selectors,
+					"AttributeSelector",
+				);
+				const pseudoClassSelectors = getSelectors(
+					selectors,
+					"PseudoClassSelector",
+				);
+				const universalSelectors = selectors.filter(
+					child =>
+						child.type === "TypeSelector" && child.name === "*",
+				);
+				const combinatorNodes = getSelectors(selectors, "Combinator");
+				const compounds = getCompounds(selectors);
+
+				const combinators = getSelectorNames(combinatorNodes);
+				const pseudoClassSelectorsNames =
+					getSelectorNames(pseudoClassSelectors);
+				const pseudoElementSelectors = getSelectors(
+					selectors,
+					"PseudoElementSelector",
+				);
+				const pseudoElementNames = getSelectorNames(
+					pseudoElementSelectors,
+				);
+				const attributeNames = attributeSelectors.map(s => s.name.name);
+				const attributeMatchers = attributeSelectors
+					.map(child => child.matcher)
+					.filter(Boolean);
+
+				if (idSelectors.length > maxIds) {
+					exceedLimitError(context, selectorLoc, maxIds, "id");
+				}
+
+				if (classSelectors.length > maxClasses) {
+					exceedLimitError(context, selectorLoc, maxClasses, "class");
+				}
+
+				if (typeSelectors.length > maxTypes) {
+					exceedLimitError(context, selectorLoc, maxTypes, "type");
+				}
+
+				if (attributeSelectors.length > maxAttributes) {
+					exceedLimitError(
+						context,
+						selectorLoc,
+						maxAttributes,
+						"attribute",
+					);
+				}
+
+				if (pseudoClassSelectors.length > maxPseudoClasses) {
+					exceedLimitError(
+						context,
+						selectorLoc,
+						maxPseudoClasses,
+						"pseudo-class",
+					);
+				}
+
+				if (universalSelectors.length > maxUniversals) {
+					exceedLimitError(
+						context,
+						selectorLoc,
+						maxUniversals,
+						"universal",
+					);
+				}
+
+				if (combinatorNodes.length > maxCombinators) {
+					exceedLimitError(
+						context,
+						selectorLoc,
+						maxCombinators,
+						"combinator",
+					);
+				}
+
+				if (compounds.length > maxCompounds) {
```

Comment:
I think by the current logic the number of compounds can be calculated by adding 1 to the number of combinators, making the `getCompounds` function unnecessary.

```suggestion
				if (combinatorNodes.length + 1 > maxCompounds) {
```